lib/puppet/functions/hiera_vault.rb

```diff
@@ -73,20 +72,24 @@ def vault_get(key, options, context)
         config.ssl_ciphers = options['ssl_ciphers'] if config.respond_to? :ssl_ciphers
       end
 
-      fail if vault.sys.seal_status.sealed?
+      if vault.sys.seal_status.sealed?
+        raise Exception, "[hiera-vault] vault is sealed"
+      end
 
       context.explain { "[hiera-vault] Client configured to connect to #{vault.address}" }
-    rescue Exception => e
+    rescue StandardError => e
       vault = nil
       context.explain { "[hiera-vault] Skipping backend. Configuration error: #{e}" }
```

Comment:
Are you sure this is what you want? For normal use it will just skip the vault data source. A user will have to use --explain, or turn on debugging to see the explain in the log. Maybe better to error. OTOH, if the rescued error is an expected error comparable to "not found", say because of referencing several optional vaults, this this may make sense. If this captures connection errors, timeouts, permissions etc. then I think a Puppet::DataBind::LookupError should be raised.